app/src/lib/get-title-bar-config.ts

```diff
@@ -0,0 +1,43 @@
+import { writeFile } from 'fs/promises'
+import { readFileSync } from 'fs'
+import { join } from 'path'
+import { app } from 'electron'
+import { TitleBarStyle } from '../ui/lib/title-bar-style'
+
+export type TitleBarConfig = {
+  titleBarStyle: TitleBarStyle
+}
+
+let cachedTitleBarConfig: TitleBarConfig | null = null
+
+// The function has to be synchronous,
+// since we need its return value to create electron BrowserWindow
+export function readTitleBarConfigFileSync(): TitleBarConfig {
+  if (cachedTitleBarConfig) {
+    return cachedTitleBarConfig
+  }
+
+  try {
+    const titleBarConfig = JSON.parse(
+      readFileSync(getTitleBarConfigPath(), 'utf8')
+    )
+
+    // Return if we found valid values
+    if (
+      titleBarConfig.titleBarStyle === 'native' ||
+      titleBarConfig.titleBarStyle === 'custom'
+    ) {
+      cachedTitleBarConfig = titleBarConfig
+      return titleBarConfig
+    }
+  } catch (error) {}
+
+  return { titleBarStyle: 'native' }
+}
+
+export async function saveTitleBarConfigFile(config: TitleBarConfig) {
```

Comment:
:art: the `async` modifier is unnecessary unless you use `await` within the function:

```suggestion
export function saveTitleBarConfigFile(config: TitleBarConfig) {
```